src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -487,9 +488,84 @@ void QgsRelationEditorWidget::deleteSelectedFeatures()
+    deletedFeaturesRequest.setSubsetOfAttributes( QgsAttributeList() << mNmRelation.referencedFields().first() );
+
+    QgsFeatureIterator deletedFeatures = layer->getFeatures( deletedFeaturesRequest );
+    QStringList deletedFeaturesPks;
+    QgsFeature feature;
+    while ( deletedFeatures.nextFeature( feature ) )
+    {
+      deletedFeaturesPks.append( QgsExpression::quotedValue( feature.attribute( mNmRelation.referencedFields().first() ) ) );
+    }
+
+    QgsFeatureRequest linkingFeaturesRequest;
+    linkingFeaturesRequest.setFlags( QgsFeatureRequest::NoGeometry );
+    linkingFeaturesRequest.setSubsetOfAttributes( QgsAttributeList() );
+
+    QString linkingFeaturesRequestExpression;
+    if ( !deletedFeaturesPks.empty() )
+    {
+      linkingFeaturesRequestExpression = QStringLiteral( "%1 IN (%2)" ).arg( QgsExpression::quotedColumnRef( mNmRelation.fieldPairs().first().first ), deletedFeaturesPks.join( ',' ) );
+      linkingFeaturesRequest.setFilterExpression( linkingFeaturesRequestExpression );
+
+      QgsFeatureIterator relatedLinkingFeatures = mNmRelation.referencingLayer()->getFeatures( linkingFeaturesRequest );
+
+      int relatedLinkingFeaturesCount = 0;
+      while ( relatedLinkingFeatures.nextFeature( feature ) )
+      {
+        relatedLinkingFeaturesCount++;
+      }
+
+      if ( deletedFeaturesPks.size() == 1 && relatedLinkingFeaturesCount > 1 )
+      {
+        QMessageBox messageBox( QMessageBox::Question, tr( "Really delete entry?" ), tr( "The entry on %1 is still linked to %2 features on %3. Do you want to delete it?" ).arg( mNmRelation.referencedLayer()->name(), QString::number( relatedLinkingFeaturesCount ), mRelation.referencedLayer()->name() ), QMessageBox::NoButton, this );
+        messageBox.addButton( QMessageBox::Cancel );
+        QAbstractButton *deleteButton = messageBox.addButton( tr( "Delete" ),  QMessageBox::AcceptRole );
+
+        messageBox.exec();
+        if ( messageBox.clickedButton() != deleteButton )
+          deleteFeatures = false;
+      }
+      else if ( deletedFeaturesPks.size() > 1 && relatedLinkingFeaturesCount > deletedFeaturesPks.size() )
+      {
+        QMessageBox messageBox( QMessageBox::Question, tr( "Really delete entries?" ), tr( "The %1 entries on %2 are still linked to %3 features on %4. Do you want to delete them?" ).arg( QString::number( deletedFeaturesPks.size() ), mNmRelation.referencedLayer()->name(), QString::number( relatedLinkingFeaturesCount ), mRelation.referencedLayer()->name() ), QMessageBox::NoButton, this );
+        messageBox.addButton( QMessageBox::Cancel );
+        QAbstractButton *deleteButton = messageBox.addButton( tr( "Delete" ), QMessageBox::AcceptRole );
+
+        messageBox.exec();
+        if ( messageBox.clickedButton() != deleteButton )
+          deleteFeatures = false;
+      }
+    }
+  }
+  else
+  {
+    layer = mRelation.referencingLayer();
+  }
+
+  if ( deleteFeatures )
+  {
+    layer->deleteFeatures( featureids );
```

Comment:
Same thing here about `layer` being a `nullptr`?

Reply:
That's actually the original code that didn't choke the last years ;P